src/SignalR/clients/java/signalr/src/test/java/com/microsoft/signalr/UserAgentTest.java

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+package com.microsoft.signalr;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.util.stream.Stream;
+
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+public class UserAgentTest {
```

Comment:
Let's also add some tests that validate the Java Vendor and Version. We'll need to read those in the test so we know what to validate against. The test should also just strip off and ignore the version info (so we don't have to keep updating it).